src/libraries/Hooks.sol

```diff
@@ -149,24 +156,43 @@ library Hooks {
         IPoolManager.ModifyLiquidityParams memory params,
         BalanceDelta delta,
         bytes calldata hookData
-    ) internal {
-        if (params.liquidityDelta > 0 && key.hooks.hasPermission(AFTER_ADD_LIQUIDITY_FLAG)) {
-            self.callHook(
-                abi.encodeWithSelector(IHooks.afterAddLiquidity.selector, msg.sender, key, params, delta, hookData)
-            );
-        } else if (params.liquidityDelta <= 0 && key.hooks.hasPermission(AFTER_REMOVE_LIQUIDITY_FLAG)) {
-            self.callHook(
-                abi.encodeWithSelector(IHooks.afterRemoveLiquidity.selector, msg.sender, key, params, delta, hookData)
-            );
+    ) internal returns (BalanceDelta hookDelta) {
+        bytes memory returnData;
```

Comment:
I think this code might be simpler if we do split out the permissions? since we wont have to handle returnData in known cases?